# Bonus: Temperature Analysis I

In [1]:
import pandas as pd
from datetime import datetime as dt

In [2]:
# "tobs" is "temperature observations"
df = pd.read_csv('Resources/hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [3]:
# Convert the date column format from string to datetime
df['date'] = pd.to_datetime(df['date'])
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19550 entries, 0 to 19549
Data columns (total 4 columns):
 #   Column   Non-Null Count  Dtype         
---  ------   --------------  -----         
 0   station  19550 non-null  object        
 1   date     19550 non-null  datetime64[ns]
 2   prcp     18103 non-null  float64       
 3   tobs     19550 non-null  int64         
dtypes: datetime64[ns](1), float64(1), int64(1), object(1)
memory usage: 611.1+ KB


In [4]:
# Set the date column as the DataFrame index
new_df = df.set_index(df.date)
new_df.head()

station       date  prcp  tobs
date                                          
2010-01-01  USC00519397 2010-01-01  0.08    65
2010-01-02  USC00519397 2010-01-02  0.00    63
2010-01-03  USC00519397 2010-01-03  0.00    74
2010-01-04  USC00519397 2010-01-04  0.00    76
2010-01-06  USC00519397 2010-01-06   NaN    73

In [5]:
# Drop the date column
new_df = new_df.drop('date', axis='columns')
new_df.head()

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73

### Compare June and December data across all years 

In [6]:
from scipy import stats

In [7]:
# Filter data for desired months
june_df = new_df.loc[new_df.index.month == 6]
december_df = new_df.loc[new_df.index.month == 12]

In [8]:
# Identify the average temperature for June
june_df.describe()
june_average_temp = june_df.tobs.mean()
print(f'The average temperature for June is: {round(june_average_temp,2)}')

The average temperature for June is: 74.94


In [9]:
# Identify the average temperature for December
december_average_temp = december_df.tobs.mean()
print(f'The average temperature for December is: {round(december_average_temp,2)}')

The average temperature for December is: 71.04


In [10]:
# Create collections of temperature data
june_list = june_df.tobs.tolist()
december_list = december_df.tobs.tolist()

In [11]:
# Run paired t-test
tt = stats.ttest_ind(june_list, december_list, equal_var=False)
tt

Ttest_indResult(statistic=31.355036920962423, pvalue=4.193529835915755e-187)

### Analysis

After comparing temperature data across all years for the months of June and December, there is a 5.5% higher average temperature in June (74.94) than in December (71.04). To reject the hypothesis of the difference being due to chance a t-test was performed, selecting a independent or unpaired one since both lists were different months of the year and had different variance. The results give a pvalue of 4.1935e-187 which is very close to 0, this means that the is statistically significant difference in the temperatures of June and December.